### PR TITLE
Simplify element names of IDs

### PR DESCRIPTION
**Pull request by ue71603**
Fixes https://github.com/VDVde/OJP/issues/181 https://github.com/VDVde/OJP/issues/179

### CODE REVIEW

**Review by AndreasAtSBB — APPROVED** [review]

**Comment by ue71603** [comment]
I will have to change all examples. And it will create major problems with other PR, that don't do it. So. I will do it asap when a lot of things have already be merged.

**Comment by skinkie** [comment]
Is there a reason not to make this an attribute?

**Comment by ue71603** [comment]
Mostly history. I can change it to an attribute.

**Review by Aurige — APPROVED** [review]

**Review by sgrossberndt — APPROVED** [review]

**Comment by ue71603** [comment]
Discussion
•	#215
•	#230
 
They both would align things more to what it should be. However, (a) we don’t do it everywhere and we should move the id to an attribute in that case (b) in breaks a lot of implementations (c) It does not really simplify anything in a heavy way. I agree: It is hygiene, but I am not sure, if we should do it.
Please advise. For now I will leave them alone

**Comment by trurlurl** [comment]
Documentation updated.

**Comment by skinkie** [comment]
Also check references so ```<LineRef>NL:Line:1</LineRef``` becomes ```<LineRef ref="NL:Line:1" />```

File: siri/acsb/acsb_accessibility-v0.3.xsd [hunk 1]
```diff
@@ -29,7 +29,7 @@
 				<Publisher>Kizoom, 109-123 Clifton Street, London EC4A 4LD</Publisher>
 				<Rights>Unclassified
        <Copyright>CEN, Crown Copyright 2006-2012</Copyright>
-    </Rights>
+				</Rights>
```
Comment (sgrossberndt) [inline siri/acsb/acsb_accessibility-v0.3.xsd:32]: This unrelated change of a siri file should be removed

File: OJP/OJP_FareSupport.xsd [hunk 2]
```diff
@@ -231,9 +231,9 @@
 			<xs:documentation>[related to FARE PRODUCT in TM and NeTEx] different FARE PRODUCTs that may be available with related information</xs:documentation>
 		</xs:annotation>
 		<xs:sequence>
-			<xs:element name="FareProductId" type="FareProductCodeType">
+			<xs:element name="FareProductCode" type="FareProductCodeType">
```
Comment (sgrossberndt) [inline OJP/OJP_FareSupport.xsd:234]: This renaming should probably be a separate pull request.
It will be forgotten in the release notes as breaking change otherwise.

File: OJP/OJP_Fare.xsd [hunk 3]
```diff
@@ -142,6 +137,11 @@
 				</xs:element>
 			</xs:choice>
 		</xs:sequence>
+		<xs:attribute name="id" type="ObjectIdType">
```
Comment (sgrossberndt) [inline OJP/OJP_Fare.xsd:140]: I know there was a reluctance to use attributes in the OJP model, I don't know why exactly, IIRC due to be able to map it easier to other data structures like JSON? @pvgrumbkow

**Review by sgrossberndt — CHANGES_REQUESTED** [review]

**Review by AndreasAtSBB — APPROVED** [review]
Changes about the Id are OK.

File: OJP/OJP_Fare.xsd (hunk 3 above)
Comment (herlitze) [reply OJP/OJP_Fare.xsd:140]: Yes, Werner confirmed that in TRIAS the mapping, besides the consistency with SIRI, was the reason to do without attributes in most cases.

Comment (ue71603) [reply OJP/OJP_Fare.xsd:140]: @herlitze @sgrossberndt Next time pls state so during the meeting (we discussed this two times and I specifically asked if I should go into that direction). Would have saved me the work. For JSON: Our guys would have prefered the attributes. I will add this to the discussion list for the next meeting.

**Review by skinkie — DISMISSED** [review]
Agree with @sgrossberndt

File: OJP/OJP_Fare.xsd (hunk 3 above)
Comment (ue71603) [reply OJP/OJP_Fare.xsd:140]: Stephan: Renaming, but element and not attribute. We want to stay consistent with Siri.

Comment (ue71603) [reply OJP/OJP_Fare.xsd:140]: adapted

File: siri/acsb/acsb_accessibility-v0.3.xsd (hunk 1 above)
Comment (ue71603) [reply siri/acsb/acsb_accessibility-v0.3.xsd]: removed

File: OJP/OJP_FareSupport.xsd (hunk 2 above)
Comment (ue71603) [reply OJP/OJP_FareSupport.xsd]: in separate

**Review by herlitze — APPROVED** [review]

**Review by AndreasAtSBB — APPROVED** [review]
Changes are OK.

File: OJP/OJP_Trips.xsd [hunk 4]
```diff
@@ -265,7 +265,7 @@
 			<xs:documentation>Structure for trip overview information.</xs:documentation>
 		</xs:annotation>
 		<xs:sequence>
-			<xs:element name="TripId" type="xs:NMTOKEN">
+			<xs:element name="TripId" type="ObjectIdType">
```
Comment (sgrossberndt) [inline OJP/OJP_Trips.xsd:268]: Is it on purpose this is still TripId and not Id?

File: OJP/OJP_Common.xsd [hunk 5]
```diff
@@ -5,6 +5,13 @@
 	</xs:annotation>
 	<xs:import namespace="http://www.siri.org.uk/siri" schemaLocation="../siri/siri_model/siri_all-v2.0.xsd"/>
 	<xs:include schemaLocation="OJP_All.xsd"/>
+	<!-- ==== Entity ================================================================ -->
```
Comment (sgrossberndt) [inline OJP/OJP_Common.xsd:8]: Either remove this line or have it as annotation documentation like "General" in line 16

**Review by sgrossberndt — CHANGES_REQUESTED** [review]
rebase necessary

File: OJP/OJP_FareSupport.xsd [hunk 6]
```diff
@@ -257,7 +257,7 @@
 		<xs:sequence>
 			<xs:element name="FareProductId" type="FareProductCodeType">
 				<xs:annotation>
-					<xs:documentation>Unique Id of the FareProduct.</xs:documentation>
+					<xs:documentation>Unique Code of the FareProduct.</xs:documentation>
```
Comment (sgrossberndt) [inline OJP/OJP_FareSupport.xsd:260]: This should only be part of https://github.com/VDVde/OJP/pull/267 I guess

File: OJP/OJP_Common.xsd (hunk 5 above)
Comment (ue71603) [inline OJP/OJP_Common.xsd:8]:
```suggestion

```

File: OJP/OJP_Trips.xsd (hunk 4 above)
Comment (ue71603) [inline OJP/OJP_Trips.xsd:268]:
```suggestion
			<xs:element name="Id" type="ObjectIdType">
```

Comment (ue71603) [reply OJP/OJP_Trips.xsd:268]: will change

File: OJP/OJP_Common.xsd (hunk 5 above)
Comment (ue71603) [reply OJP/OJP_Common.xsd:8]: will change

File: OJP/OJP_FareSupport.xsd (hunk 6 above)
Comment (ue71603) [reply OJP/OJP_FareSupport.xsd:260]: forgot to remove it

Comment (ue71603) [inline OJP/OJP_FareSupport.xsd:260]:
```suggestion
					<xs:documentation>Unique Id of the FareProduct.</xs:documentation>
```

**Review by AndreasAtSBB — APPROVED** [review]
Changes about the Id look correct and examples are also fixed. (Just made an optical check, did'n run them.)

**Review by Aurige — DISMISSED** [review]

**Review by sgrossberndt — APPROVED** [review]

**Review by skinkie — APPROVED** [review]

**Comment by trurlurl** [comment]
Question: As the type of those Ids were changed from xs:NMTOKEN to ObjectIdType - shouldn't we change the following as well?

TripSummaryStructure.TripId
TripFareProductReferenceStructure.FromTripIdRef, .ToTripIdRef, .FromTripLegIdRef, .ToTripLegIdRef
TripFareResultStructure.FromTripLegIdRef, .ToTripLegIdRef
(Perhaps more)

**Comment by skinkie** [comment]
Sounds reasonable, also from a validating perspective.
